src/pages/ListScholarships.jsx

```diff
@@ -58,6 +79,23 @@ function ListScholarships() {
     pruneQueryParam(qParams.MAX_AMOUNT);
   }
 
+  /**
+   * prunes invalid grade values not respresented by GradeLevel enum
+   */
+  if (Array.isArray(grades) && JSON.stringify(grades) !== '[]') {
+    const prunedInvalid = [...new Set(grades)].filter((g) =>
+      GradeLevel.keys().includes(g)
+    );
+    if (prunedInvalid.length !== grades.length) {
+      replaceQueryParam(qParams.GRADES, prunedInvalid);
+    }
+  }
+  if (
+    (grades !== undefined && !Array.isArray(grades)) ||
+    JSON.stringify(grades) === '[]'
+  )
```

Comment:
Surround both `if` statements with blocks with the `undefined` and `isArray` checks so that these other conditions are simplified. E.g.

```js
if (grades !== undefined) {
  if (!Array.isArray(grades)) {
    // prune
  }
  // validate/prune array values
  // param altogether if it's empty.
}
```

Reply:
Use `grade.length` to check for emptiness